src/amen.coffee

```diff
@@ -1,19 +1,21 @@
 require "colors"
 
-timeout = (t, promise) ->
+race = (t, promise) ->
   timer = new Promise (_, reject) ->
     setTimeout (-> reject new Error "Async test timed out"), t
   Promise.race [ timer, promise ]
 
+timeout = (t, promise) -> if t then race t, promise else promise
+
 # TODO: use explicit result objects, instead of true | Error | undefined
-test = (description, definition) ->
+test = (description, definition, timeLimit=1000) ->
```

Comment:
Once we add that third argument, this starts getting cryptic. Let's go ahead and add an alternative, where we can add options argument instead.

```
test = (description, definition) -> 
  if description.constructor == Object
    {description, wait} = description
  else if description.constructor == String
    {wait} = defaults
  else
    description = description.toString()
  if definition.call?
    # ...
```

Reply:
Of course, this assumes we've defined `defaults` somewhere above. :)

Reply:
I've also left out the possibility of setting a timer on the parent of nested tests because that's a bit confusing. Let's keep it simple, as in your implementation.

Reply:
The idea here is be able to write:

```coffee
test
  description: "A simple test"
  wait: 1000 # 1 second
  -> true 
```

Also note that this is not only clearer (in terms of what the `100` refers to) but also allows us to add more options later if we choose.